File: de.lmu.ifi.dbs.knowing.core/src/de/lmu/ifi/dbs/knowing/core/graph/xml/Property.java
package de.lmu.ifi.dbs.knowing.core.graph.xml;

import javax.xml.bind.annotation.XmlAttribute;

public class Property {
	
	//@XmlAttribute
	private String key;

	//@XmlAttribute
	private String value;

	public Property() {
		this("","");
	}
	
	public Property(String key) {
		this(key, "");
	}

	public Property(String key, String value) {
		this.key = key;
		this.value = value;
	}
	
	@XmlAttribute
	public String getKey() {
		return key;
	}

	public void setKey(String key) {
		this.key = key;
	}

	@XmlAttribute
	public String getValue() {
		return value;
	}

	public void setValue(String value) {
		this.value = value;
	}
	
	
}

File: de.lmu.ifi.dbs.knowing.core/src/de/lmu/ifi/dbs/knowing/core/graph/xml/Property.scala
package de.lmu.ifi.dbs.knowing.core.graph.xml

import scala.annotation.target.field
import javax.xml.bind.annotation.{ XmlAccessorType, XmlAccessType, XmlAttribute }

@XmlAccessorType(XmlAccessType.FIELD)
class Property(@(XmlAttribute @field) var key: String,
  @(XmlAttribute @field) var value: String) {

  def this() = this("", "")

  override def toString(): String = key + " => " + value

}